driver/node/opera.go

```diff
@@ -203,6 +209,22 @@ func (n *OperaNode) StreamLog() (io.ReadCloser, error) {
 }
 
 func (n *OperaNode) Stop() error {
+	// Send ctrl+c to signal client termination
+	n.Interrupt()
+
+	// Wait until client terminate
+	// if not enough, artifacts will be corrupted after export
+	time.Sleep(3 * time.Second)
+
+	// Signal to listeners that client is terminated.
+	// All listener calls is expected to be blocking.
+	n.listenerMutex.Lock()
+	for listener := range n.listeners {
```

Comment:
what is benefit of these listeners. Why the node just cannot call `ExportGenesis`, `ExportEvents` here?

It looks to me that each node calls its own methods `ExportGenesis`, `ExportEvents` via a system of listeners. I think calling these method directly (when export is configured) would simplify the code a lot

Reply:
As a default behavior, the nodes do not export anything. It only exports if the scenario tells it to.

The listeners enable us to attach only to targeted nodes when we populate events in executor.
It also fits the pattern we used for network as well.

Is there a downside to using listeners that I didn't think about?

Reply:
the listeners are generalyl used when there is 1:N or M:N relation. For instance one (1)  network informs all (N) nodes about an update. 

But here, if I get the code right, does that 1 node, informs itself to execute an action - i.e. is equivalent of exectuing the action directly. 

I understand that export may happen only when configured.